### DISCUSSION TITLE
Can I plug the raspberry pi to the board without using usb?

### DISCUSSION THREAD

**Discussion by supercaro** [bassamanator/Sovol-SV06-firmware#12] [Q&A]
Hi friends.

The micro usb port seems broken on my SV06 mainboard and I was wondering if it was possible to connect the pi to one port inside instead. I know it's possible with some Big Tree Tech board but I can't seem to find anything about it online. I can get a new board but I have accumulated a good amount of rubber duckies already lol.

Thanks a lot friends and thanks to bassamanator for their hard work. You're a champ. :)

**Comment by bassamanator** [comment]
I'm looking at the board and unfortunately I don't see what interface would allow communication to a Rpi or other device. If you can find a BTT board that allows for this, then we can look for the initerface on this Sovol board.

**Comment by supercaro** [comment]
Thank you for your fast answer. After extensive diagnostics, we found a short in the board as well so I will be using a BTT SKR mini e3 v2 as a replacement. I am grateful for your answer and hopefully it'll help someone that might have the same question later on.

**Reply by bassamanator** [reply]
Great!

Please consider creating a pull request when you have the printer.cfg ready for that board. You can also just paste it here so that others might benefit from it.

Best!

**Reply by supercaro** [reply]
I don't know much about how to make configs for klipper (or about github), but i frankensteined your profiles with the one recommanded for the btt skr mini e3 v2 and my old sv06 profile (which i also think was yours). It seems to work here but might not be a bad idea, if someone wants to use it, to double check everything themselves to make sure it won't break their printer.

[include ./cfgs/misc-macros.cfg]
[include ./cfgs/PARKING.cfg]
[include ./cfgs/MECHANICAL_GANTRY_CALIBRATION.cfg]
[include ./cfgs/beeper.cfg]

# NOTE Uncomment the ONE of the following lines if you're using an adxl345
# [include ./cfgs/adxl-rp2040.cfg]
# [include ./cfgs/adxl-direct.cfg]

[mcu]
# NOTE Obtain definition by "ls -l /dev/serial/by-id/"
serial: /dev/ttyAMA0
baud: 300000
restart_method: command


[printer]
kinematics: cartesian
max_velocity: 180
max_accel: 800
max_z_velocity: 10
max_z_accel: 120

#####################################################################
#      X/Y/Z Stepper Settings
#####################################################################

[stepper_x]
step_pin: PB13
dir_pin: PB12
enable_pin: !PB14
microsteps: 64
rotation_distance: 40
endstop_pin: tmc2209_stepper_x:virtual_endstop
position_endstop: 0
position_max: 225 # NOTE You can adjust this if you have more room
homing_speed: 40
homing_retract_dist: 0

[tmc2209 stepper_x]
uart_pin: PC11
tx_pin: PC10
run_current: 0.580
stealthchop_threshold: 999999
interpolate: False
sense_resistor: 0.150
uart_address: 0
driver_SGTHRS: 75
diag_pin: PC0

[stepper_y]
step_pin: PB10
dir_pin: !PB2
enable_pin: !PB11
microsteps: 64
rotation_distance: 40
endstop_pin: tmc2209_stepper_y:virtual_endstop
position_endstop: 0
position_max: 225 # NOTE You can adjust this if you have more room
homing_speed: 40
homing_retract_dist: 0

[tmc2209 stepper_y]
uart_pin: PC11
tx_pin: PC10
run_current: 0.580
stealthchop_threshold: 999999
interpolate: False
sense_resistor: 0.150
uart_address: 2
driver_SGTHRS: 75
diag_pin: PC1

[stepper_z]
step_pin: PB0
dir_pin: PC5
enable_pin: !PB1
microsteps: 64
rotation_distance: 4
endstop_pin: probe:z_virtual_endstop
position_min: -4
position_max: 258 # NOTE You can adjust this if you have more room
homing_speed: 5

[tmc2209 stepper_z]
uart_pin: PC11
tx_pin: PC10
run_current: 0.580
stealthchop_threshold: 999999
interpolate: False
sense_resistor: 0.150
uart_address: 1
# diag_pin: PC2 # Should not be needed

#####################################################################
#   Extruder
#####################################################################

[extruder]
step_pin: PB3
dir_pin: PB4
enable_pin: !PD2
microsteps: 32
rotation_distance: 4.65
nozzle_diameter: 0.400
filament_diameter: 1.750
heater_pin: PC9
sensor_type: EPCOS 100K B57560G104F
sensor_pin: PA0
#control: pid
#pid_kd: 86.991
#pid_kp: 21.479
#pid_ki: 1.326
min_temp: 0
max_temp: 300
max_extrude_only_distance: 150.0

[tmc2209 extruder]
uart_pin: PC11
tx_pin: PC10
run_current: 0.659
stealthchop_threshold: 999999
interpolate: False
sense_resistor: 0.150
uart_address: 3

#####################################################################
#   Bed Heater
#####################################################################

[heater_bed]
heater_pin: PC8
sensor_type: EPCOS 100K B57560G104F
sensor_pin: PC3
#control: pid
#pid_kp: 64.440
#pid_ki: 0.773
#pid_kd: 1343.571
min_temp: 0
max_temp: 110

#####################################################################
#   Probe Related
#####################################################################

[probe]
pin: PC14
x_offset: 27
y_offset: -20
z_offset: 0
samples: 3
samples_result: median
samples_tolerance: 0.01
samples_tolerance_retries: 5

[safe_z_home]
home_xy_position: 84.50, 135 # NOTE The probe is at the center of the bed.
speed: 100.0
z_hop: 10
z_hop_speed: 15

[bed_mesh]
speed: 175
mesh_min: 28, 28
mesh_max: 205, 205
probe_count: 5,5
algorithm: bicubic
fade_start: 1
fade_end: 10
fade_target: 0

#####################################################################
#   Fans
#####################################################################

[fan]
pin: PC6

[controller_fan electronic_enclosure_fan]
pin: PB15
idle_timeout: 60

[heater_fan hotend_fan]
pin: PC7
heater: extruder
heater_temp: 50.0

#####################################################################
#   Misc
#####################################################################

[display]
lcd_type: st7920
cs_pin: PB8
sclk_pin: PB9
sid_pin: PD6
encoder_pins: ^PA10, ^PA9
click_pin: ^!PA15

[bed_screws]
screw1: 0.5, 51.5
screw1_name: front left
screw2: 170.1, 52.5
screw2_name: front right
screw3: 170.2, 221.5
screw3_name: back right
screw4: 0.5, 221.5
screw4_name: back left

[screws_tilt_adjust]
screw1: 85, 136.5
screw1_name: center
screw2: 0.1, 51.5
screw2_name: front left
screw3: 170.1, 51.5
screw3_name: front right
screw4: 170.1, 221.5
screw4_name: back right
screw5: 0.1, 221.5
screw5_name: back left
speed: 50
horizontal_move_z: 10
screw_thread: CCW-M4

[gcode_macro screws_adjust]
gcode:
    M117 Tramming...
    SCREWS_TILT_CALCULATE
  G90
  G1 z120 F500
  G1 X125 F500

[gcode_macro disable_steppers]
gcode:
    M18 

[exclude_object]

**Comment by bassamanator** [comment]
@supercaro Thanks for this. Just to confirm, this is the `printer.cfg` you are using with the SV06 with the SKR-Mini-E3-V2.0 board?

**Reply by supercaro** [reply]
yes :)

**Reply by alexdelli** [reply]
I have SKR-Mini-E3-V2.0 board, is it worth to change with V3 ?

**Reply by bassamanator** [reply]
@alexdelli Not worth it.
